server/cronworkflow/cron_workflow_server.go

```diff
@@ -34,10 +48,20 @@ func (c *cronWorkflowServiceServer) ListCronWorkflows(ctx context.Context, req *
 	if req.ListOptions != nil {
 		options = *req.ListOptions
 	}
-	return auth.GetWfClient(ctx).ArgoprojV1alpha1().CronWorkflows(req.Namespace).List(options)
+	return auth.GetWfClient(ctx).ArgoprojV1alpha1().CronWorkflows(req.Namespace).List(c.withInstanceID(options))
 }
 
 func (c *cronWorkflowServiceServer) CreateCronWorkflow(ctx context.Context, req *cronworkflowpkg.CreateCronWorkflowRequest) (*v1alpha1.CronWorkflow, error) {
+	if c.mode == GRPCServerMode && len(c.instanceID) > 0 {
```

Comment:
should also work for kube client - should not need the guard condition

Reply:
Same reply as `server/workflow/workflow_server.go` :)